boa/src/syntax/ast/token.rs

```diff
@@ -42,6 +42,17 @@ impl Debug for VecToken {
     }
 }
 
+/// Represents the type differenct types of numeric literals.
+#[cfg_attr(feature = "serde-ast", derive(Serialize, Deserialize))]
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum NumericLiteral {
+    /// A floating point number
+    Number(f64),
+    /// A integer
+    Integer(u64),
+    // TODO: Add BigInt
```

Comment:
How big is a `BigInt`?

Reply:
BigInt can hold any arbitrary large integer. All the numerical are implemented in the software level.

[num-bigint](https://crates.io/crates/num-bigint) crate